code_challenges.rb

```diff
@@ -20,15 +38,75 @@
 album3 = 'Abbey Road'
 # Expected output: 'bby Rd'
 
+# Create a method called vowel_removed
+# Takes a parameter of a String
+# Iterate over the string using a block
+# Check each value against a list of values and if it is a value then it will be deleted
+# Return string with out vowels
+
+def vowel_removed string
+  string.delete! "aeiouAEIOU"
+  p string
+end
+
+vowel_removed album3
 
+# I have realized javascript makes you overthink and get into every single tiny detail. It's a little refreshing and a little scary that you can take all of that logic/thought process and condense it down to just a simple "delete this value". I initially forgot that I needed to put the ! at the end to make it mutate the string. ALSO forgot to account for uppercase letters. Glad I ran through all three strings.
 
 # --------------------3a) Create a class called Bike that is initialized with a model, wheels, and current_speed. The default number of wheels is 2. The current_speed should start at 0. Create a get_info method that returns a sentence with all the data from the bike object.
 
-# Expected output example: 'The Trek bike has 2 wheels and is going 0 mph.'
+# I need to create a class called Bike
+# I need to initialize it with mode, wheels and current speed
+# I need to set the wheels to 2 and the speed to 0
+# I need to create a method called get_info and have it output a string that consists of string interpolation to grab the model, wheels and current speed from the initializer.
+# Assign a new instance of the class to object trek_bike
+# call trek_bike and get_info to print out string.
+
+class Bike
+  def initialize model
+    @model = model
+    @wheels = 2
+    @current_speed = 0
+  end
+  def get_info
+    "The #{@model} bike has #{@wheels} wheels and is going #{@current_speed} mph."
+  end
+end
 
+trek_bike = Bike.new "Trek"
+p trek_bike.get_info
+# Expected output example: 'The Trek bike has 2 wheels and is going 0 mph.'
 
 
 # -------------------3b) Add the ability to pedal faster and brake. The pedal_faster method should increase the speed. The brake method should decrease the speed. The bike cannot go negative speeds.
 
 # Expected output example: my_bike.pedal_faster 10 => 10
 # Expected output example: my_bike.brake 15 => 0
+
+class Bike
+  def initialize model
+    @model = model
+    @wheels = 2
+    @current_speed = 0
+  end
+  def get_info
+    "The #{@model} bike has #{@wheels} wheels and is going #{@current_speed} mph."
+  end
+  def pedal_faster
+    @current_speed += 10
+  end
+  def brake
+    if @current_speed >= 15
+       @current_speed -= 15
+    elsif @current_speed < 15
+          @current_speed = 0
```

Comment:
Okay quick question should braking always be a static number or should we let users choose how much they want to brake? What does that imply?

Reply:
You know it wasn't until we were going over it that I realized I mis read the instructions. I thought we needed to use the example numbers as our set inputs and outputs, then upon rereading understood that a dynamic solution was requested.